File: backend/src/SecurityService_Core/Models/ControllerDTO/Administrator/AdminChangeInputModel.cs
﻿using System.ComponentModel.DataAnnotations;

namespace SecurityService_Core.Models.ControllerDTO.Administrator
{
    public class AdminChangeInputModel
    {
        [Required]
        [Length(6, 100)]
        public string UserName { get; set; }
        [Required]
        [Length(6, 50)]
        [EmailAddress]
        public string Email { get; set; }
        [Required]
        [Length(11, 20)]
        [Phone]
        public string PhoneNumber { get; set; }
        [Required]
        [Length(6, 100)]
        public string FIO { get; set; }
        [Required]
        [Length(10, 100)]
        public string Organization { get; set; }
        [Required]
        public string INN { get; set; }
        [Required]
        public int Role { get; set; }
        [Required]
        public int State { get; set; }
        [Required]
        [MinLength(10), MaxLength(256)]
        public string Address { get; set; }
    }
}

File: frontend/src/app/interfaces/adminChangeInputModel.ts
import { UserRoles } from "../enums/userRoles";
import { UserStates } from "../enums/userStates";

export interface AdminChangeInputModel {
    userName: string;
    email: string;
    phoneNumber: string;
    fio: string;
    organization: string;
    inn: string;
    role: number;
    state: number;
    address: string;
}